src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -92,6 +92,25 @@ public static string GetDirectoryName(string path)
             return null;
         }
 
+        public static ReadOnlySpan<char> GetDirectoryName(ReadOnlySpan<char> path)
+        {
+            if (path.IsEmpty)
```

Comment:
This is not the same as `IsEffectivelyEmpty`. You should make an overload of `IsEffectivelyEmpty` that takes a ROS so you can call it.

Reply:
I mean
```c#
bool IsEffectivelyEmpty(string path)
{
     return IsEffectivelyEmpty(path.AsReadOnlySpan);
}
```

Reply:
Or eliminate `IsEffectivelyEmpty(string path)` and make all its callers do `.AsReadOnlySpan()` thus slowly spanifying more and more of the BCL internals up to just below where strings come in from public API's. Is that a reasonable direction @jkotas? Seems like `.AsReadOnlySpan()` is very cheap. Or would that end up too much ugliness

Reply:
> Is that a reasonable direction @jkotas?

Yes.

>  Or would that end up too much ugliness

You do not actually need to write `.AsReadOnlySpan` yourself. There is implicit `string` -> `ReadOnlySpan<char>` conversion.

Reply:
Oh yes - I guess you would only need it to resolve overloads.